tools/integration-tests/tester/tests/consensus/consensus_conflict_spam_test.go

```diff
@@ -0,0 +1,185 @@
+package consensus
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/iotaledger/hive.go/crypto/ed25519"
+
+	"github.com/iotaledger/goshimmer/packages/consensus/gof"
+	"github.com/iotaledger/goshimmer/packages/jsonmodels"
+	"github.com/iotaledger/goshimmer/packages/ledgerstate"
+	"github.com/iotaledger/goshimmer/tools/integration-tests/tester/framework"
+	"github.com/iotaledger/goshimmer/tools/integration-tests/tester/tests"
+	"github.com/stretchr/testify/require"
+)
+
+// constant var, shouldn't be changed
+var tokensPerRequest int
+
+/**
+TestConflictSpam spams a node with conflicts and makes sure the GoFs are the same across the network
+*/
+func TestConflictSpam(t *testing.T) {
+	ctx, cancel := tests.Context(context.Background(), t)
+	defer cancel()
+	n, err := f.CreateNetwork(ctx, t.Name(), 4, framework.CreateNetworkConfig{
+		Faucet:      true,
+		StartSynced: true,
+		Activity:    true,
+	}, tests.EqualDefaultConfigFunc(t, false))
+	require.NoError(t, err)
+	defer tests.ShutdownNetwork(ctx, t, n)
+
+	faucet, peer1 := n.Peers()[0], n.Peers()[1]
+	tokensPerRequest = faucet.Config().TokensPerRequest
+
+	tests.AwaitInitialFaucetOutputsPrepared(t, faucet, n.Peers())
+
+	fundingAddress := peer1.Address(0)
+	tests.SendFaucetRequest(t, peer1, fundingAddress)
+	require.Eventually(t, func() bool {
+		return tests.Balance(t, peer1, fundingAddress, ledgerstate.ColorIOTA) >= uint64(tokensPerRequest)
+	}, tests.Timeout, tests.Tick)
+
+	addresses := make([]*ledgerstate.Address, 100)
```

Comment:
Why Address pointers? Do you plan to modify them from another function?

Reply:
I am kinda of always passing pointers as part of a java mindset that I still have. It is just cheaper to copy them then the actual structs.
I guess for this script it doesn't matter whether we use memory wisely and I can remove the pointers because it is maybe more readable\safer...

My question is what do we do for goshimmer? If we play it safe and not pass transactions by pointers then we bloat memory and strain the GC. If we pass pointers we are less safe since tx are supposed to be immutable. 
I would wrap them in an immutable struct that only exports getters if goshimmer would be production ready. I guess since goshimmer isn't production ready we are just doing the safe approach which is simple and bloaty?

Reply:
Really want to know your thoughts on that.

In the meanwhile I changed the addresses to be non-pointers.
By the way I tried to do the same for transactions but I got warning since they have mutexes.

Reply:
In go, all types are passed via value (int, bool, float and so on); variable of a certain type though, do not contain the entire value itself, but just a reference to it: making them effectively always pointers: this is the case for slices and maps. If you pass a slice to a function and you modify an element of it, the change would be reflected in the caller too. You can then consider that all "complex" types are passed implicitly by pointers. The only exception to this are struct themselves: if you pass a struct as a parameter, a full copy of it will be passed (non-recursively though, e.g. complex types within a struct will point to the same memory location). Hence why golang allows you to declare methods of structs using the base struct as is or using a pointer notation: the latter is used if the fields of the calling struct will be modified by the method itself as part of the operation.

In your case, declaring the inner elements of the slice as pointers doesn't really change anything: unless what you want is to change something on the inner elements themselves!

I hope I didn't make this explanation too confusing!